src/NetPanzer/Interfaces/MapInterface.cpp

```diff
@@ -55,11 +55,11 @@ bool MapInterface::startMapLoad( const char *file_path, const char *mapstyle_mpa
     NPString stpath = mapstyle_mpath;
     NPString tsname = main_map.getAssocTileSet();
 
-    char subbuff[24];
-    memcpy( subbuff, &tsname[0], 23); // max 23 chars for tileset name
-    char * pch;
-    pch = strstr (subbuff,".tls");
-    strncpy (pch,"\0",1);
+    const int max_len_tsname = 23;
+    std::string subbuff = tsname.substr(0, max_len_tsname);
+    ssize_t found = subbuff.find(".tls");
+    if (found != -1)
```

Comment:
nit: maybe can compare against [string::npos](https://cplusplus.com/string::npos) directly

Reply:
Looks like I should definitely do that https://stackoverflow.com/a/15740842